Formula/gemstash.rb

```diff
@@ -0,0 +1,124 @@
+class Gemstash < Formula
+  desc "RubyGems.org cache and private gem server"
+  homepage "https://github.com/bundler/gemstash"
+  url "https://rubygems.org/downloads/gemstash-2.0.0.gem"
+  sha256 "511a5773c43507b913c4f2f57d33a3d972bd71378b2669c22a357b3351a82bf4"
+  head "https://github.com/bundler/gemstash.git"
+
+  if MacOS.version < :mojave
+    depends_on "openssl"
+    depends_on "ruby"
+  end
+
+  def install
+    if build.head?
+      system "gem", "build", "gemstash.gemspec"
+      gem_file = Dir["*.gem"].first
+    else
+      gem_file = "gemstash-#{version}.gem"
+    end
+
+    ENV["GEM_HOME"] = libexec
+
+    if MacOS.version < :mojave
+      gem = Formula["ruby"].bin/"gem"
+    else
+      gem = "gem"
+    end
+
+    system gem, "install", "--no-document", gem_file
+
+    (bin/"gemstash").write_env_script libexec/"bin/gemstash",
+      :GEM_HOME                      => ENV["GEM_HOME"],
+      :SINATRA_ACTIVESUPPORT_WARNING => false
+
+    (var/"gemstash").mkpath
+    (etc/"gemstash").mkpath
+    config_file.write <<~EOS
+      ---
+      :base_path: "#{var}/gemstash"
+      :bind: tcp://127.0.0.1:#{port}
+      :cache_type: memory
+      :db_adapter: sqlite3
+      :protected_fetch: false
+      :fetch_timeout: 20
+    EOS
+  end
+
+  def caveats; <<~EOS
```

Comment:
This seems like something that isn't homebrew specific and should therefore be in the upstream manual